app/models/indicator.rb

```diff
@@ -6,6 +6,27 @@ class Indicator < ApplicationRecord
 
   validates :name, presence: true
 
+  class << self
+    def fetch_all(order_options)
+      order_direction = order_options['order_direction'].present? && order_options['order_direction'] == 'desc' ? :desc : :asc
+
+      case order_options['order_type']
+        when 'category'
+          indicators = Indicator.order(category: order_direction, name: :asc)
+        when 'stack_family'
+          indicators = Indicator.order(stack_family: order_direction, name: :asc)
+        when 'definition'
+          indicators = Indicator.order(definition: order_direction, name: :asc)
+        when 'unit'
+          indicators = Indicator.order(unit: order_direction, name: :asc)
+        else
+          indicators = Indicator.order(name: order_direction)
+      end
```

Comment:
Maybe you could simplify by splitting into 1. param whitelisting step and 2. order clause construction step. This would allow you to avoid the big `case`. For the whitelisting step you could do sth like `order_type = (['category', 'stack_family', other permitted cols].include?(order_options['order_type']) && order_options['order_type']`; then if `order_type` is not nil yo can construct the order clause as `{order_type => order_direction, name: :asc}`, otherwise just `{name: :asc}` - this should shut up hound :)